components/channel_view/channel_view.jsx

```diff
@@ -60,46 +47,6 @@ export default class ChannelView extends React.PureComponent {
         );
     }
 
-    updateChannel(props) {
-        let channel;
-        const fakechannel = (new URLSearchParams(props.location.search)).get('fakechannel');
-        if (fakechannel) {
-            channel = JSON.parse(fakechannel);
-        } else {
-            channel = ChannelStore.getByName(props.match.params.channel);
-
-            if (channel && channel.type === Constants.DM_CHANNEL) {
-                loadNewDMIfNeeded(channel.id);
-            } else if (channel && channel.type === Constants.GM_CHANNEL) {
-                loadNewGMIfNeeded(channel.id);
-            }
-
-            if (channel) {
-                GlobalActions.emitChannelClickEvent(channel);
-            } else {
```

Comment:
Does removing `updateChannel` will not cause any side-effect?
Why not just get the channel name from `props.match.params` and use it accordingly?
```
    const {team, identifier} = props.match.params;
    const channelName = identifier.toLowerCase();
```

Reply:
It looks like it may break the "fake channel" logic, but I don't actually know if that's been used for a long time. We had it in there for when we'd show random users in your DM list on first sign up even if the channels for them didn't exist so that we could create those channels when you did click on them.

I was actually wondering yesterday if we can just remove all that logic since I think that was the only place we used them, and we haven't automatically added DM users for a long time

Reply:
ok. I will remove fake channel logic and retest.

Reply:
Can we touch base about this @hmhealey so I can ensure that I understand what you are suggesting.

Reply:
I think we could just make a ticket and go with these changes if they work. The fake channel logic is split across the redux and web app, and it might even be in the mobile app even though that logic was removed probably 2 years ago

Reply:
Thanks for clearing up!